pkg/pagerduty/pagerduty.go

```diff
@@ -36,14 +35,15 @@ func NewWithToken(authToken string) (PagerDuty, error) {
 
 // New will create a PagerDuty struct with all of the required fields
 func New(client *sdk.Client) (PagerDuty, error) {
-	user, err := getCurrentUser(client)
+	sdkUser, err := getCurrentUser(client)
 	if err != nil {
-		return PagerDuty{}, pkgerrors.Wrap(err, "could not retrieve the current user")
+		return PagerDuty{}, fmt.Errorf("could not retrieve the current user: %w", err)
 	}
 
+	user := toLocalUser(sdkUser)
 	resp := PagerDuty{
-		c:         client,
-		userEmail: user.Email,
+		c:           client,
+		currentUser: user,
```

Comment:
@ramonbutter this is the one-way conversion I have as of now. is that fine for now?

Reply:
it's ok, you can define any data type you want. I always wanted to write as little code as possible. But I also see that in some cases there is also a gain in redefining things with limited scope to decouple the business logic from the external SDK.